src/main/java/com/segment/analytics/android/integrations/intercom/IntercomIntegration.java

```diff
@@ -272,6 +272,7 @@ private void setUserAttributes(Traits realTraits, @Nullable Map<String, Object>
   private Company setCompany(Map<String, Object> payload) {
     Company.Builder company = new Company.Builder();
     company.withCompanyId(String.valueOf(payload.get("id")));
```

Comment:
what if payload is null?

Reply:
An id is required to associate a user with a company, so if a customer doesn't include this, they're not following directions. ;)

That said, I can add an early return if the id doesn't exist to prevent a null pointer exception.

Reply:
It won't lead to a null pointer exception; it'll just set company Id to `null`.

For the future I would point out that that `.id` is not a requirement for our spec, it's an optional field. So we should make sure to guard against that case when applicable (doesn't apply here as much though since it won't cause a NPE).

Reply:
Gotcha re: null pointer.

Right, `id` isn't part of our spec, but Intercom won't create or associate a user with a company if a companyId isn't present. I think we shouldn't rely solely on *our* spec to determine whether we guard against a null.

I just added an early return if `id` isn't present for efficiency's sake.

Reply:
LGTM.